### ISSUE TITLE
Debugger Stop button has to be clicked twice

### ISSUE THREAD

**Issue by john-aws** [microsoft/vscode#96730]
- VSCode Version: 1.44.2
- OS Version: OS X 10.14.6

Steps to Reproduce:

1. create a JavaScript launch target in launch.json
2. set a breakpoint in the code (so that it doesn't immediately run to completion)
3. click Start Debugging
4. wait for breakpoint to be hit, pausing execution
5. click the debugger's Stop button

I would expect the debugger to stop the running process and closed the floating debugger menubar, but neither happens. When I click the Stop button a second time it works as expected. This is 100% reproducible. I wonder if this might be a feature, not a bug, but I don't see any configuration items related to this behavior and I don't know why this button would have to be clicked twice when none of the others are.

Does this issue occur when all extensions are disabled?: Yes

**Comment by isidorn** [comment]
This is beacuse the first stop sends a `disconnect` to the extension, the second stop click is a hard stop.
It seems like you are debugging javascript, thus forwarding this to @roblourens and @connor4312 since the debug extensions seems to ignore the `disconnect`

**Comment by connor4312** [comment]
@john-aws can you enable `"trace": true` in the launch configuration and send us a log?

/needsMoreInfo

**Comment by john-aws** [comment]
@connor4312 Yes, I can collect a log file. Is it the file named `debugadapter-legacy.txt`?

And how should I share this with you? It's a 2000-line text file. I doubt you want it inline in this issue. Thanks.

**Comment by connor4312** [comment]
https://gist.github.com/ is a good place for it!

**Comment by john-aws** [comment]
Having reviewed the log file's contents, it's too invasive so I won't be able to supply it. If there are key things you are looking for, I'd be happy to provide those. I'm assuming you are unable to repro this with a stock JavaScript application.

**Comment by connor4312** [comment]
Hey there. Since filing this issue we've moved to a new debugger, and done a lot of fixes and improvements. Does this still happen today?
